File: Leetcode Daily Challenge/025-July-2025/014-convert-binary-number-in-a-linked-list-to-integer.java
// https://leetcode.com/problems/convert-binary-number-in-a-linked-list-to-integer/



// Definition for singly-linked list.
public class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}

class Solution {
    public int getDecimalValue(ListNode head) {
        int decimal = 0;
        ListNode temp = head;
        while (temp != null) {
            decimal = (decimal * 2) + temp.val;
            temp = temp.next;
        }
        return decimal;
    }
}

File: Leetcode Daily Challenge/025-July-2025/014-convert-binary-number-in-a-linked-list-to-integer.cpp
// https://leetcode.com/problems/convert-binary-number-in-a-linked-list-to-integer/




#include <bits/stdc++.h>
using namespace std;

// Definition for singly-linked list.
struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution {
public:
    int getDecimalValue(ListNode* head) {
        int decimal = 0;
        ListNode* temp = head;
        while (temp) {
            decimal = (decimal * 2) + temp -> val;
            temp = temp -> next;
        }
        return decimal;
    }
};